src/azure-cli/azure/cli/command_modules/appconfig/_azconfig/utils.py

```diff
@@ -135,13 +135,21 @@ def __parse_connection_string(connection_string):
     return endpoint, id_, secret
 
 
-def __encode_keyword(string):
-    if six.PY2:
-        # python 2 compatible
-        return None if string is None else urllib.quote(string, safe='')  # pylint: disable=E1101
+def __unescape_encode_keyword(string):
+    if string is not None:
+        import ast
+
+        # ast library requires quotes around string
+        string = '"{0}"'.format(string)
+        string = ast.literal_eval(string)
```

Comment:
Why do you add a  ast.literal_eval() here?

Reply:
We are trying to unescape the input argument. Our service supports unicode as a key or value. For those unprintable characters such as tab and null. Client can escape it and use it in CLI command. 

---
In reply to: [347080887](https://github.com/Azure/azure-cli/pull/11036#discussion_r347080887) [](ancestors = 347080887)